subsys/pm/pm.c

```diff
@@ -238,8 +245,8 @@ bool pm_system_suspend(int32_t ticks)
 	pm_state_set(z_cpus_pm_state[id].state, z_cpus_pm_state[id].substate_id);
 	pm_stats_stop();
 
+#if defined(CONFIG_PM_DEVICE) && DT_PM_DEVICE_NEEDED
 	/* Wake up sequence starts here */
-#if defined(CONFIG_PM_DEVICE)
 	if (atomic_add(&_cpus_active, 1) == 0) {
 		pm_resume_devices();
```

Comment:
One small issue I noticed testing this today:

The `pm_resume_devices()` call here may still run when the low power state we're returning from doesn't have the `pm_device_enabled` flag set, and thus `pm_resume_devices()` above wasn't called.
So in DT configurations where not all low power states have said flag, we'll be executing device pm resume logic unnecessarily for devices that hadn't been suspended.

Maybe we could locally retain the `z_cpus_pm_state[id].pm_device_enabled` bool check above and use that here to only resume devices that have actually been suspended by system pm?

Reply:
You're right of course.
I should've checked the device resume function more closely.